modules/api-import-export/src/main/java/org/wso2/carbon/apimgt/importexport/utils/APIImportUtil.java

```diff
@@ -368,6 +368,158 @@ public static void importAPI(String pathToArchive, String currentUser, boolean i
+            importedApi.setStatus(targetApi.getStatus());
+        } catch (IOException e) {
+            String errorMessage = "Error in reading API definition. ";
+            log.error(errorMessage, e);
+            throw new APIImportException(errorMessage, e);
+        } catch (APIManagementException e) {
+            String errorMessage = "Error in checking API existence. ";
+            log.error(errorMessage, e);
+            throw new APIImportException(errorMessage, e);
+        }
+
+        Set<Tier> allowedTiers;
+        Set<Tier> unsupportedTiersList;
+
+        try {
+            allowedTiers = provider.getTiers();
+        } catch (APIManagementException e) {
+            String errorMessage = "Error in retrieving tiers of the provider. ";
+            log.error(errorMessage, e);
+            throw new APIImportException(errorMessage, e);
+        }
+
+        if (!(allowedTiers.isEmpty())) {
+            unsupportedTiersList = Sets.difference(importedApi.getAvailableTiers(), allowedTiers);
+
+            //If at least one unsupported tier is found, it should be removed before adding API
+            if (!(unsupportedTiersList.isEmpty())) {
+                for (Tier unsupportedTier : unsupportedTiersList) {
+
+                    //Process is continued with a warning and only supported tiers are added to the importer API
+                    log.warn("Tier name : " + unsupportedTier.getName() + " is not supported.");
+                }
+
+                //Remove the unsupported tiers before adding the API
+                importedApi.removeAvailableTiers(unsupportedTiersList);
+            }
+        }
+
+        try {
+            //Swagger definition will only be available of API type HTTP. Web socket api does not have it.
+            if (!APIConstants.APIType.WS.toString().equalsIgnoreCase(importedApi.getType())) {
+
+                String swaggerContent = FileUtils.readFileToString(
+                        new File(pathToArchive + APIImportExportConstants.SWAGGER_DEFINITION_LOCATION));
+
+                addSwaggerDefinition(importedApi.getId(), swaggerContent);
+
+                //Load required properties from swagger to the API
+                Set<URITemplate> uriTemplates = definitionFromOpenAPISpec.getURITemplates(importedApi, swaggerContent);
+                importedApi.setUriTemplates(uriTemplates);
+                Set<Scope> scopes = definitionFromOpenAPISpec.getScopes(swaggerContent);
+                importedApi.setScopes(scopes);
+
+                for (URITemplate uriTemplate : uriTemplates) {
+                    Scope scope = uriTemplate.getScope();
+                    if (scope != null && !(APIUtil.isWhiteListedScope(scope.getKey()))) {
+                        if (provider.isScopeKeyAssigned(importedApi.getId(), scope.getKey(),
+                                PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId(true))) {
```

Comment:
`PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId(true)` can internally cause a database call. Try to avoid that by calling that once in the top, assign to a variable and reuse that.

Reply:
noted, will fix that